src/axom/quest/examples/quest_distributed_distance_query_example.cpp

```diff
@@ -459,13 +496,79 @@ class ObjectMeshWrapper
     using PointType = primal::Point<double, DIM>;
     using PointArray = axom::Array<PointType>;
 
+    // compute start and stop angle, allowing for some empty ranks
+    double thetaStart, thetaEnd;
+    int numPoints = 0;
+
+    // perform scan on ranks to compute numPoints, thetaStart and thetaEnd
+    {
+      int hasPoints = rankHasPoints ? 1 : 0;
+      int myRank = m_mesh.getRank();
+      int numRanks = m_mesh.getNumRanks();
+
+      axom::Array<int> arr(numRanks, numRanks);
+      arr.fill(-1);
+      MPI_Allgather(&hasPoints, 1, MPI_INT, arr.data(), 1, MPI_INT, MPI_COMM_WORLD);
+
+      SLIC_DEBUG_IF(
+        m_verbose,
+        axom::fmt::format("After all gather: [{}]", axom::fmt::join(arr, ",")));
+
+      axom::Array<int> sums(numRanks + 1, numRanks + 1);
+      sums[0] = 0;
+      for(int i = 1; i <= numRanks; ++i)
+      {
+        sums[i] = sums[i - 1] + arr[i - 1];
+      }
```

Comment:
Note: This is a scan, but the distributed closest point algorithm doesn't require RAJA, so I couldn't use the RAJA version.

Reply:
RAJA scans should be used for OpenMP, CUDA, and HIP, correct?

Reply:
True, but this one's just on the host -- it's scanning a single scalar per rank -- so not really worth the transfer cost, or requiring a strict dependency on RAJA for the query.